### DISCUSSION TITLE
How to properly create sections in marked?

### DISCUSSION THREAD

**Discussion by 4e576rt8uh9ij9okp** [markedjs/marked#2464] [Q&A]
Hello,
so the knowledge I have is that the right structure of HTML is as follows:

```html
<article>
    <h1>About cars</h1>
    <section>
        <h2>Engine Types</h2>
        <p>Some text</p>
    </section>
    <section>
        <h2>Car types</h2>
        <p>Some text</p>
    </section>
</article>
```

is there an option for this?

**Comment by UziTech** [comment]
There is no section element in markdown but you can use HTML in markdown. Your example in markdown would be:

```markdown
<article>

# About cars

<section>

## Engine Types

Some text

</section>
<section>

## Car types

Some text

</section>
</article>

```

(The blank lines are needed before and after markdown content otherwise marked will think it is part of the HTML and leave it alone.)

**Reply by 4e576rt8uh9ij9okp** [reply]
That's good to know but could this be maybe an optional function to tell marked to create the sections?

{ sections: true }

**Reply by UziTech** [reply]
Adding a new option for everything everyone wants will make marked unusable.

![image](https://user-images.githubusercontent.com/97994/167272825-d173e088-7347-4601-bdbd-4d41869dd6e5.png)

**Reply by 4e576rt8uh9ij9okp** [reply]
The only difference is that I asked for a proper HTML structure, and not to "color" the headers or some other stupid function.

**Reply by UziTech** [reply]
Ya the problem is that you aren't the only one asking for another option. Since there is already a way to do it, and as far as I can tell you are the only one asking for that feature, I don't see a reason to maintain it in the library.

You could create an [extension](https://marked.js.org/using_pro#extensions) that does this.

If this is something you think should be standard you would have to change the [CommonMark spec](https://github.com/commonmark/commonmark-spec).
